CONTRIBUTING.md

```diff
@@ -249,62 +261,66 @@ $ yarn docs:api
 
 ### Coding style
 
-Please follow the coding style of the project. MUI uses prettier and eslint, so if possible, enable linting in your editor to get real-time feedback.
+Please follow the coding style of the project.
+MUI Core projects use prettier and eslint, so if possible, enable linting in your editor to get real-time feedback.
 
 - `yarn prettier` reformats the code.
-- `yarn eslint` runs manually the linting rules.
+- `yarn eslint` runs the linting rules.
 
-Finally, when you submit a Pull Request, they are run again by our continuous integration tools, but hopefully, your code is already clean!
+When you submit a PR, these checks are run again by our continuous integration tools, but hopefully your code is already clean!
 
-## How to add a new demo in the documentation
+## How to add a new demo to the documentation
 
-If, for example, you want to add new demos for the button component, you have to take the following steps:
+The following steps explain how to add a new demo to the docs using the Button component as an example:
 
-### 1. Add a new React component file under the related directory
+### 1. Add a new component file to the directory
 
-In this case, you are going to add the new file to the following directory:
+Add the new file to the component's corresponding directory...
 
 ```bash
 docs/src/pages/components/buttons/
 ```
 
-and give it a name: `SuperButtons.js`.
+...and give it a name: how about `SuperButtons.tsx`?
+
+### 2. Write the demo code
+
+MUI Core uses TypeScript to document our components.
+We prefer demos written in TS (using the `.tsx` file format).
+
+After creating a TypeScript demo, run `yarn docs:typescript:formatted` to automatically create the JavaScript version, which is also required.
+
+If you're not familiar with TypeScript, you can write the demo in JavaScript, and a core contributor may help you migrate it to TS.
 
-### 2. Edit the page Markdown file
+### 3. Edit the page's Markdown file
 
-The Markdown file is the source for the website documentation. So, whatever you wrote there will be reflected on the website.
-In this case, the file you need to edit is `docs/src/pages/components/buttons/buttons.md`.
+The Markdown file in the component's respective folder—in this case, `/buttons/buttons.md`—is the source of content for the document.
+Any changes you make there will be reflected on the website.
 
-Changes should only be applied to the English version e.g. only `app-bar.md` and
-not `app-bar-de.md`. For contributions concerning translations please read the [section
-about translations](#translations).
+Note that you should only edit the English version of this doc—do not edit `buttons-pt.md` or `buttons-zh.md`.
```

Comment:
I think all available `*-pt` and `*-zh` will be gone with https://github.com/mui/material-ui/pull/39221 — so, do we need to mention them if they don't exist? 🤔

Reply:
Good catch - I missed this on the latest draft!